subprojects/build-cache-http/src/integTest/groovy/org/gradle/caching/http/internal/HttpBuildCacheServiceTest.groovy

```diff
@@ -93,16 +93,16 @@ class HttpBuildCacheServiceTest extends Specification {
         destFile.bytes == content
     }
 
-    def "can cache artifact with redirect"() {
-        def destFile = tempDir.file("cached.zip")
+    def "storing to cache does not follow redirects"() {
         def content = "Data".bytes
         server.expectPutRedirected("/cache/${key.hashCode}", "/redirect/cache/${key.hashCode}")
-        server.expectPut("/redirect/cache/${key.hashCode}", destFile, HttpStatus.SC_OK, null, content.length)
 
         when:
         cache.store(key, writer(content))
         then:
-        destFile.bytes == content
+        BuildCacheException exception = thrown()
+
+        exception.message == "Received redirect (302) to ${server.uri}/redirect/cache/${key.hashCode} when loading entry from '${server.uri}/cache/${key.hashCode}'"
```

Comment:
Can we say `HTTP 302`?

Reply:
I think we should also explain that "this is not allowed because it's bad for performance" or something.

Reply:
I changed it to `HTTP 302` and added an explanation.